pkg/generator/ingress_translator_test.go

```diff
@@ -719,6 +719,89 @@ func TestIngressTranslatorWithHTTPOptionDisabled(t *testing.T) {
 	}
 }
 
+// TestIngressTranslatorHTTP01Challenge
+func TestIngressTranslatorHTTP01Challenge(t *testing.T) {
+	test := struct {
+		name  string
+		in    *v1alpha1.Ingress
+		state []runtime.Object
+		want  *translatedIngress
+	}{
+		name: "http01-challenge",
+		in:   ingHTTP01Challenge("simplens", "simplename"),
+		state: []runtime.Object{
+			svcHTTPO1Challenge("simplens", "cm-acme-http-solver"),
```

Comment:
`svc()` and `svcHTTPO1Challenge` do not have big diff, so we can still use `svc` with optional func like;

```suggestion
		       svc("simplens", "cm-acme-http-solver", func(svc *corev1.Service) {
			       svc.Spec.Ports = []corev1.ServicePort{{
				       Name:	   "http01-challenge",
				       TargetPort: intstr.FromInt(8089),
			       }}
		       }),
```